src/product/product.service.ts

```diff
@@ -70,6 +77,30 @@ export class ProductService {
     });
   }
 
+  async findProductByCategoryId(categoryId: number) {
+    const category = await this.categoryService.findOne(categoryId);
+
+    if (!category) {
+      throw new NotFoundException("Category not found");
+    }
+    const parentId: number = categoryId;
+    const subCategoryId = await this.subCategoryService.findOneSubCategory(parentId);
+
+    if (!subCategoryId) {
+      throw new NotFoundException("SubCategory not found");
+    }
+    const subCategories = await this.subCategoryService.searchSubCategories(parentId);
```

Comment:
I don't see the utility of the 'findOneSubCategory' function above as in this line you're fetching all the subcategory related to a specific category then we can directly apply the check whether a sub category is present or not on the basis of this response. 
I can't see the utility of the above API call as the check can be done with the response of this API as well.
Please provide an explanation if the above API call is necessary else refactor the code and optimise.